.github/instructions/sdk-generation.instructions.md

```diff
@@ -0,0 +1,129 @@
+**Message to user**: "TypeSpec validation takes around 20 - 30 seconds."
+**Actions**:
+1. Run `azsdk_run_typespec_validation` to validate the TypeSpec project.
+2. If validation succeeds, proceed to Step 4
+3. If validation fails:
+    - Display all compilation errors to user
+    - Agent should provide suggestions to fix them and prompt the user to verify the fixes. If agent cannot resolve the errors, then prompt the user to fix compilation errors"
+    - Wait for user to fix errors and re-run validation. Provide detailed information about all the changes done by copilot and prompt the user before rerunning the validation.
+**Success Criteria**: TypeSpec compilation passes without errors
+
+Step 4: Review and Commit Changes
+**Goal**: Stage and commit TypeSpec modifications
+**Condition**: Only if the TypeSpec validation succeeds (from Step 3)
+**Actions**:
+1. Run `azsdk_get_modified_typespec_projects` to identify changes
+2. If no changes found, inform: "No TypeSpec projects were modified in current branch" and go to SDK generation mentioned in [typespec to sdk](../../eng/common/instructions/azsdk-tools/typespec-to-sdk.instructions.md).
+3. Display all modified files (excluding `.github` and `.vscode` folders)
+4. Prompt user: "Please review the modified files. Do you want to commit these changes? (yes/no)"
+5. If yes:
+    - If on main branch, prompt user: "You are currently on the main branch. Please create a new branch using `git checkout -b <branch-name>` before proceeding."
+    - Wait for user confirmation before continuing
+    - Run `git add <modified-files>`
+    - Prompt for commit message
+    - Run `git commit -m "<user-provided-message>"`
+    - Run `git push -u origin <current-branch-name>`
+**Success Criteria**: Changes committed and pushed to remote branch
+
+Step 5: Create Specification Pull Request
+**Goal**: Create PR for TypeSpec changes if not already created
+**Condition**: Only if there are committed changes (from Step 4)
+**Actions**:
+1. Prompt the user to confirm if a pull request already exists for API spec changes. If answer is no or unsure then check if spec PR already exists using `azsdk_get_pull_request_link_for_current_branch`
+2. If PR exists, display PR details and proceed to next steps in SDK generation.
+3. If no PR exists:
+    - Inform user: "No pull request found for the current branch. Proceeding to create a new pull request."
+    - Create a pull request using `azsdk_create_pull_request_for_current_branch`
+    - Prompt for PR title and description
+    - Display PR creation progress
+    - Wait for PR creation confirmation
+    - Display created PR details
+4. Inform the user to follow the instructions on spec PR to get approval from API reviewers and merge the spec PR.
+**Success Criteria**: Specification pull request exists
+
+Follow the steps in [typespec to sdk](../../eng/common/instructions/azsdk-tools/typespec-to-sdk.instructions.md) to generate the SDK.
```

Comment:
Do we plan to merge this PR to main currently? I don't see the instructions in the linked file mentioning that the release intent scenario would use the pipeline for generation. `update-version` MCP tool isn't fully ready for release scenario yet.